hedera-node/src/main/java/com/hedera/services/context/NodeInfo.java

```diff
@@ -0,0 +1,55 @@
+package com.hedera.services.context;
+
+import com.swirlds.common.AddressBook;
+
+import java.util.function.Supplier;
+
+/**
+ * Summarizes useful information about the nodes in the {@link AddressBook}
+ * from the Platform. In the future, there may be events that require
+ * re-reading the book; but at present nodes may treat the initializing
+ * book as static.
+ */
+public class NodeInfo {
+	private boolean bookIsRead = false;
+
+	private int n;
+	private boolean[] isZeroStake;
+
+	private final Supplier<AddressBook> book;
+
+	public NodeInfo(Supplier<AddressBook> book) {
+		this.book = book;
+	}
+
+	/**
+	 * Returns true if the given id refers to a missing node, or a
+	 * node in the address book with explicit zero stake.
+	 *
+	 * @param nodeId the id of interest
+	 */
+	public boolean isZeroStake(long nodeId) {
+		if (!bookIsRead) {
+			readBook();
+		}
+
+		final int index = (int)nodeId;
+		if (index < 0 || index >= n) {
```

Comment:
I think index < 0 is something severe that might need a throw.

Reply:
Agreed, we should _never_ actually request info for a missing node...changed to `throw` in this case.